semgrep-core/src/core/ast/AST_generic.ml

```diff
@@ -362,7 +370,7 @@ and id_info = {
    * a typed entity, which can be interpreted as a TypedMetavar in semgrep.
    * alt: have an explicity type_ field in entity.
    *)
-  id_type : type_ option ref;
+  id_type : type_ option ref; [@equal fun _a _b -> true]
```

Comment:
This might be a holdover from a previous PR, but basically we are failing to unify metavariables at some point because of the fact that we compare types (probably due to a bug in type inference).

I would be inclined to leave this in, knowing that it will prevent roadblocks down the line.

Reply:
If this causes problems we should have a reproducible test case.